collector-sqs/src/main/java/zipkin/collector/awssqs/AwsSqsCollector.java

```diff
@@ -0,0 +1,196 @@
+import zipkin.storage.StorageComponent;
+
+
+public class AwsSqsCollector implements CollectorComponent, Closeable {
+
+  private static final Logger log = LoggerFactory.getLogger(AwsSqsCollector.class);
+
+  public static final class Builder implements CollectorComponent.Builder {
+
+    Collector.Builder delegate = Collector.builder(AwsSqsCollector.class);
+
+    List<String> queueUrls;
+    int waitTimeSeconds = 20; // aws sqs max wait time is 20 seconds
+    AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
+    int parallelism = 1;
+
+    @Override public CollectorComponent.Builder storage(StorageComponent storageComponent) {
+      delegate.storage(storageComponent);
+      return this;
+    }
+
+    @Override public CollectorComponent.Builder metrics(CollectorMetrics metrics) {
+      delegate.metrics(Util.checkNotNull(metrics, "metrics").forTransport("sqs"));
+      return this;
+    }
+
+    @Override public CollectorComponent.Builder sampler(CollectorSampler sampler) {
+      delegate.sampler(sampler);
+      return this;
+    }
+
+    /** SQS queue URLs to consume from */
+    public Builder queueUrls(List<String> queueUrls) {
+      this.queueUrls = queueUrls;
+      return this;
+    }
+
+    /** AWS credentials for authenticating calls to SQS. */
+    public Builder credentialsProvider(AWSCredentialsProvider credentialsProvider) {
+      this.credentialsProvider = credentialsProvider;
+      return this;
+    }
+
+    /** Amount of time to wait for messages from SQS */
+    public Builder waitTimeSeconds(int seconds) {
+      this.waitTimeSeconds = seconds;
+      return this;
+    }
+
+    /** How many processors to run in parallel for each queue URL */
+    public Builder parallelism(int parallelism) {
+      this.parallelism = parallelism;
+      return this;
+    }
+
+    @Override public CollectorComponent build() {
+      return new AwsSqsCollector(this);
+    }
+
+    Builder() {
+    }
+  }
+
+  private final LazyAmazonSQSAsync client;
+  private final LazyProcessors processors;
+
+  AwsSqsCollector(Builder builder) {
+    client = new LazyAmazonSQSAsync(builder.credentialsProvider);
+    processors = new LazyProcessors(client.get(), builder.delegate.build(), builder.queueUrls,
+        builder.parallelism, builder.waitTimeSeconds);
+  }
+
+  @Override public CollectorComponent start() {
+    processors.get();
+    return this;
+  }
+
+  @Override public CheckResult check() {
+    try {
+      client.get(); // make sure compute doesn't throw an exception
+      return processors.check(); // check if any processor has failed
+    } catch (RuntimeException e) {
+      return CheckResult.failed(e);
+    }
+  }
+
+  @Override public void close() throws IOException {
+    client.close();
+    processors.close();
+  }
+
+  private static final class LazyProcessors extends LazyCloseable<List<AwsSqsSpanProcessor>>
+      implements Component {
+
+    private final List<String> queueUrls;
```

Comment:
note. if you private the class (as you've done), there's no need to private fields. they look less cluttered without visibility modifiers (and the bytecode will be smaller)